src/fparser/Fortran2003.py

```diff
@@ -268,38 +287,87 @@ class BlockBase(Base):
 ----------
 content : tuple
     """
+    @staticmethod
     def match(startcls, subclasses, endcls, reader,
-              match_labels = False,
-              match_names = False, set_unspecified_end_name = False,
-              match_name_classes = (),
-              enable_do_label_construct_hook = False,
-              enable_if_construct_hook = False,
-              enable_where_construct_hook = False,
-              enable_select_type_construct_hook = False,
-              enable_case_construct_hook = False
-              ):
+              match_labels=False,
+              match_names=False,
+              set_unspecified_end_name=False,
+              match_name_classes=(),
+              enable_do_label_construct_hook=False,
+              enable_if_construct_hook=False,
+              enable_where_construct_hook=False,
+              enable_select_type_construct_hook=False,
+              enable_case_construct_hook=False):
+        '''
+        Checks whether the content in reader matches the given
+        type of block statement (e.g. DO..END DO, IF...END IF etc.)
+
+        :param type startcls: The class marking the beginning of the block
+        :param list subclasses: List of classes that can be children of
+                                the block
+        :param type endcls: The class marking the end of the block
+        :param reader: Content to check for match
+        :type reader: str or instance of :py:class:`FortranReaderBase`
+        :param bool match_labels: TBD
```

Comment:
Lots of TBD entries here. Please either justify why or update.

Reply:
I'd adopted the principle of just documenting what I understood and leaving anything I didn't (on the basis that there was no interface doc at all before).